client/src/components/MainMap.jsx

```diff
@@ -77,10 +66,25 @@ export default function MainMap() {
         {popupRender && (
             <>
             <p>LAT: {popupRender[1]} LON: {popupRender[0]}</p>
-            <ReportForm/>
+            <button onClick={() => setReportModal(true)}>Report</button>
+            {reportModal && (
+                <ReportForm onClose={() => setReportModal(false)}/>
+            )
+            }
             </>
         )}
     </div>
 </div>
 );
 }
+
+const popupStyle = {
```

Comment:
in react, using css classes vs inline styles is better for performance! style objects can cause unecessary re-renders, and makes it almost impossible to use shared styling. Id recommend moving this to a .css file :)